gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -193,7 +194,7 @@ public String toString() {
 
   private volatile boolean isActive = false;
 
-  public DagManager(Config config, boolean instrumentationEnabled) {
+  public DagManager(Config config, JobStatusRetriever jobStatusRetriever, boolean instrumentationEnabled) {
```

Comment:
@Singleton?

Reply:
It already is. "@singleton" annotation is placed on the class, and that controls Guice behavior. We also select the constructor that will be used for dependency injection with "@Inject" annotation.